src/interfaces.ts

```diff
@@ -17,6 +17,8 @@ export type ConceptStatusType =
   | 'TRANSLATED'
   | 'UNPUBLISHED';
 
+export type AvailabilityType = 'everyone' | 'teacher' | 'student';
```

Comment:
Bør sjekke om dette funker.
```suggestion
export enum AvailabilityType {
EVERYONE='everyone',
TEACHER='teacher',
STUDENT='student',
;
```

Reply:
Hva får vi ut av det?

Reply:
>For meg er det greit at AvailabilityType er definert i interface, men dette ser ut som duplisering. Hadde det funka med nokke sånt?
const availabilityValues: AvailabilityType[] = [AvailabilityType.EVERYONE, AvailabilityType.TEACHER, AvailabilityType.STUDENT]; Men det krever jo at AvailabilityType er en enum.

Type literals i typescript fungerer på samme måte som enums, jeg hadde ikke fått lov å skrive noe annet enn de 3 strengene definert i AvailabilityTypes, for da hadde typescript klaget at strengen er ikke støttet av typen. Men kan godt gjøre det dersom vi går for denne måten å skrive literals på

Reply:
Sjekket og enig med Gunnar, fungerer bra i isolate cases som i LearningResourceForm eller TopicArticleForm. Går for enums

Reply:
Se min andre kommentar

Reply:
Må ikkje bruke enums. Tenkte det berre gav færre plasser å oppdatere verdier.

Reply:
Du må vel fortsatt oppdatere både enumen og lista

Reply:
Sant, men du unngår eventuelle skrivefeil i lista.

Reply:
> const availabilityValues = ['everyone', 'teacher', 'student'] as const;
> type AvailabilityType = typeof availabilityValues[number];

Foretrekker denne løsningen, men enig at den ikke vil hjelpe i filer uten ts. Så om det blir nødvendig å ta hensyn til så kan jeg gi meg.

Reply:
https://stackoverflow.com/questions/40275832/typescript-has-unions-so-are-enums-redundant#:~:text=As%20far%20as%20I%20see,the%20resulting%20javascript%20(sample).&text=With%20the%20recent%20versions%20of,to%20declare%20iterable%20union%20types.
Fant en diskusjon her, med litt forskjellige argumenter

Reply:
Ombestemmer meg tilbake til ikke enum løsningen. Ser at det er kun 2 js filer som kunne ha bruk av enumen så det er ikke verdt å bytte når man må håndtere det annerledes alle andre steder i etterkant